src/transformers/models/seaformer/convert_seaformer_original_to_pytorch.py

```diff
@@ -0,0 +1,208 @@
+    if size == "L":
+        config.num_encoder_blocks = 3
+        config.depths = [3, 3, 3]
+        config.channels = [32, 64, 128, 192, 256, 320]
+        config.cfgs = [
+                [   [3, 3, 32, 1],  
+                    [3, 4, 64, 2], 
+                    [3, 4, 64, 1]],  
+                [
+                    [5, 4, 128, 2],  
+                    [5, 4, 128, 1]],  
+                [
+                    [3, 4, 192, 2],  
+                    [3, 4, 192, 1]],
+                [
+                    [5, 4, 256, 2]],  
+                [
+                    [3, 6, 320, 2]]
+            ]
+        config.emb_dims = [192, 256, 320]
+        config.key_dims = [16, 20, 24]
+        config.num_heads = 8
+        config.mlp_ratios = [2, 4, 6]
+        config.in_channels = [128, 192, 256, 320]
+        config.in_index = [0, 1, 2, 3]
+        config.decoder_channels = 192
+        config.embed_dims = [128, 160, 192]
+        config.hidden_sizes = [128]
+
+    else:
+        raise ValueError(f"Size {size} not supported")
+
+    # load feature extractor (only resize + normalize)
+    feature_extractor = SeaformerFeatureExtractor(
+        image_scale=(512, 512), keep_ratio=False, align=False, do_random_crop=False
+    )
+
+    # prepare image
+    image = prepare_img()
+    pixel_values = feature_extractor(images=image, return_tensors="pt").pixel_values
+
+    logger.info(f"Converting model {model_name}...")
+
+    # load original state dict
+    state_dict = torch.load(checkpoint_path, map_location=torch.device("cpu"))["state_dict"]
+
+    # rename keys
+    state_dict = rename_keys(state_dict, encoder_only=encoder_only)
+
+    # create HuggingFace model and load state dict
+    model = SeaformerForSemanticSegmentation(config)
+    model.load_state_dict(state_dict)
+    model.eval()
+
+    # forward pass
+    outputs = model(pixel_values)
+    logits = outputs.logits
+
+    # set expected_slice based on model name
+    # ADE20k checkpoints
+    if model_name == "SeaFormer_L_bs32_43.8_ade":
+        expected_slice = torch.tensor(
+            [   
+                [[ -2.2633,  -4.8691,  -6.1536], [ -3.3708,  -7.4803,  -8.9196], [ -2.9856,  -8.2059, -10.0520]],
+                [[ -5.5072,  -7.3183,  -8.4885], [ -6.2318,  -8.9810, -10.4004], [ -6.0988,  -9.3973, -11.3405]],
+                [[ -9.4179, -11.6332, -13.3330], [-10.4780, -14.1904, -16.1810], [-10.4400, -14.9166, -17.2796]],
+            ]
+        )
+
+    # verify logits
+    assert logits.shape == expected_shape
+    print(logits[0, :3, :3, :3])
+    assert torch.allclose(logits[0, :3, :3, :3], expected_slice, atol=1e-2)
+
+    # finally, save model and feature extractor
+    logger.info(f"Saving PyTorch model and feature extractor to {pytorch_dump_folder_path}...")
+    Path(pytorch_dump_folder_path).mkdir(exist_ok=True)
+    model.save_pretrained(pytorch_dump_folder_path)
+    feature_extractor.save_pretrained(pytorch_dump_folder_path)
+
```

Comment:
Could you also add a push_to_hub argument to upload the model to the hub?

It's really convenient as it takes care of repo initialization too:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/maskformer/convert_maskformer_swin_to_pytorch.py#L302